SignalServiceKit/src/Messages/OWSMessageManager.m

```diff
@@ -1442,6 +1442,13 @@ - (void)finalizeIncomingMessage:(TSIncomingMessage *)incomingMessage
         return;
     }
 
+    if (envelope.type == SSKProtoEnvelopeTypeFriendRequest) {
+        [thread setFriendRequestStatus:TSThreadFriendRequestStatusRequestReceived withTransaction:transaction];
+        incomingMessage.isFriendRequest = YES;
+    } else if (incomingMessage.body == @"") { // Assumed to be an accept friend request message
```

Comment:
This isn't a good assumption to make.

The flow also doesn't take into account the case where we sent a friend request and received one.

Refer to the desktop messenger - `js/models/messages.js:handleDataMessage`
```js
if (message.get('type') === 'friend-request') {
            if (conversation.hasSentFriendRequest()) {
              // Automatically accept incoming friend requests if we have send one already
              autoAccept = true;
              message.set({ friendStatus: 'accepted' });
              await conversation.onFriendRequestAccepted();
              window.libloki.api.sendFriendRequestAccepted(
                message.get('source')
              );
            } else if (!conversation.isFriend()) {
              await conversation.onFriendRequestReceived();
            }
          } else {
            await conversation.onFriendRequestAccepted();
          }
```

Reply:
`conversation.onFriendRequestAccepted()` just sets the status to `friends` if `conversation.hasSentFriendRequest()` is true.

After that it updates the messages and stuff to show friend request accepted.

Reply:
Ah yeah, fair enough